### PR TITLE
Milestone 4: Add Basic Interface

### PR DESCRIPTION
**Pull request by RichardChileya**
### In this section, I :
- [ ] Created a new branch
  - The interface branch holds the app's basic interface
- [ ] Created an app.RB file that will serve as your console app entry-point which has the following methods:
   - List all books.
   - List all people.
- [ ] Created a person (teacher or student, not a plain Person).
   - Created a book.
   - Created a rental.
   - List all rentals for a given person's id.
- [ ] Created the Main File that does: 
  - Present the user with a list of options to perform.
  - Lets users choose an option and parameters ask for the option.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    option = gets.chomp.to_i
+    case option
+    when 1..6 then option_func(option)
+    when 7 then puts 'Thank you for using this app'
+    else puts 'please enter a valid number'
+    end
+  end
```
Comment (muneebulrehman) [inline app.rb]: You did a great job with the project. However, there is this issue, when I enter the wrong number which is not in the options, it quits when it should loop back to the menu. I recorded [this video](https://www.loom.com/share/56963a8f90a34f13aad5424761ea2613) where I demonstrated the problem. Kindly fix this issue. This is a part of the project requirements.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      if rental.person.id == id
+        puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+      else
+        puts 'No rental ID found:'
+      end
+    end
```
Comment (muneebulrehman) [inline app.rb:161]: Optional. 
Whenever I list the rentals, there is a random output of  **No rental ID found:** even though the rentals exist. Have a look at the below screenshot and try to fix this issue. ⬇️
image

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @RichardChileya , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @RichardChileya , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**